### DISCUSSION TITLE
Overriding the ShowMailController::class

### DISCUSSION THREAD

**Discussion by spoyntersmith** [themsaid/laravel-mail-preview#81] [Ideas]
Hey, I was wondering if I could override the ShowMailController::class. I'm happy to open a PR if it's something you're happy to accept? I could add this to the config file and `registerRouteMacro()` in the MailPreviewServiceProvider

I'm looking to add access to attachments and display the headers in the above/below the view.

**Comment by freekmurze** [comment]
I don't think to many people need this customisation, so let's keep the package simple for now.

If you want to use another controller, I think it might be best to manually register the route (do more or less then same as what happens in the route macro now).

**Reply by spoyntersmith** [reply]
Thanks @freekmurze, that's a simpler way to achieve the override
